pilot/pkg/networking/core/v1alpha3/cluster_waypoint.go

```diff
@@ -148,31 +148,29 @@ func (cb *ClusterBuilder) buildWaypointInboundVIP(proxy *model.Proxy, svcs map[h
 	return clusters
 }
 
+// TODO(ilrudie) this is a bit broken by removing sa scoping as a concept. need to rework it
+// we now calculate a list of all sa's that are served by a waypoint based on waypoint network address, consider using that
```

Comment:
TBH I'm fine with a nonexact match - matching on an _exact_ SPIFFEID isn't necessary anywhere we do it, and we don't need to go out of our way to require it - just makes it harder for us to extend or customize the SPIFFE ID.